tests/test_coaddBase.py

```diff
@@ -0,0 +1,134 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+import unittest.mock
+
+import numpy as np
+
+import lsst.utils.tests
+
+import lsst.afw.image
+import lsst.daf.persistence
+from lsst.pipe.tasks.coaddBase import CoaddBaseTask, CoaddBaseConfig
+
+
+class CoaddBaseTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        np.random.seed(10)
+
+        self.config = CoaddBaseConfig()
+        self.task = CoaddBaseTask(self.config, "testCoaddBase")
```

Comment:
These aren't tests of `MakeCoaddTempExp`, they're tests of `CoaddBase`. I don't see how `CoaddBase` could be an ABC anyway: it has no unimplemented methods, so it seems like a perfectly legitimate class (as are `Task` and `CmdLineTask`). If they are supposed to be ABCs, we need to define the missing methods and what their API is.

Reply:
The implicit unimplemented method is `run()`.   `getCalExp()` probably should have been in `makeCoaddTempExp` anyway.  It's logically part of `makeCoaddTempExp` and `assembleCoadd` doesn't use it. 

After the conversation below about modifying task configs and
> I and others have used this system in tests before: I wasn't aware there was anything odd about it.

I feel even more strongly about leaving anti-examples out of unittests.  You can leave it if you but a big scary inline comment warning people to never use coaddBaseTask  because it doesn't do anything and only exists to reduce code duplication between assemble and makeCoaddTempExp.

Reply:
If `getCalExp()` should live in `makeCoaddTempExp`, I'm happy to fix that right now, and then change the test appropriately. Is there anything else in the `CoaddBase` interface that shouldn't live in the base class?

Note that `CoaddBase` cannot be an ABC with `run()` as the unimplemented interface, because `MakeCoaddTempExp` and `AssembleCoadd` have different call signatures for `run()`.

We cannot just declare something as "don't instantiate this object" unless we actually make that true in the code.

Reply:
That works for me